app/services/middleware_topology_service.rb

```diff
@@ -33,14 +36,14 @@ def entity_display_type(entity)
 
   def build_entity_data(entity)
     data = build_base_entity_data(entity)
-    data.merge!(:status => 'Unknown',
-                :display_kind => entity_display_type(entity))
-    data[:icon] = entity.decorate.try(:listicon_image)
+    data[:status] = 'Unknown'
+    data[:display_kind] = entity_display_type(entity)
+    data[:icon] = entity.decorate.try(:listicon_image) unless entity.kind_of?(MiddlewareDatasource)
```

Comment:
I've done it in a way that the fonticon from the decorator is honored, but the UTF char is taken from the `UiServiceMixin`, there are new helper methods on this module for the mapping